libethereum/Client.h

```diff
@@ -334,7 +334,7 @@ class Client: public ClientBase, protected Worker
     Block m_working;                        ///< The state of the client which we're sealing (i.e. it'll have all the rewards added), while we're actually working on it.
     BlockHeader m_sealingInfo;              ///< The header we're attempting to seal on (derived from m_postSeal).
     bool remoteActive() const;              ///< Is there an active and valid remote worker?
-    bool m_remoteWorking = false;           ///< Has the remote worker recently been reset?
+    std::atomic<bool> m_remoteWorking = { false };          ///< Has the remote worker recently been reset?
```

Comment:
I don't see why this is needed, could you post the TSAN warning about this, if there's any?

Reply:
```
Test Case "jsonrpc_isListening": 
==================
WARNING: ThreadSanitizer: data race (pid=15795)
  Read of size 1 at 0x7da800004d70 by thread T27:
    #0 dev::eth::Client::doWork(bool) /home/wins/Ethereum/cpp2/cpp-ethereum/libethereum/Client.cpp:697 (testeth+0x000000be71bf)
Previous write of size 1 at 0x7da800004d70 by main thread:
    #0 dev::eth::Client::onPostStateChanged() /home/wins/Ethereum/cpp2/cpp-ethereum/libethereum/Client.cpp:585 (testeth+0x000000bd3c8e)
```

Reply:
weird, this test only turns on/off the network, not clear why `OnPostStateChange` is called
Does it not show the rest of the stack?

Reply:
Ok, let's leave this as atomic, changing `ClientTests` doesn't seem easy